File: src/cli/team/get.ts
import type { Command } from "commander";
import { getClient } from "../../lib/client.ts";
import { formatEstimateScale, getValidEstimates } from "../../lib/estimates.ts";
import { printError, printJson } from "../../lib/output.ts";
import { resolveTeam } from "../../lib/resolvers.ts";

export function registerGet(team: Command): void {
  team
    .command("get")
    .description("Team details and members")
    .argument("<id>", "Team ID or key")
    .action(async (id: string) => {
      try {
        const client = getClient();
        const t = await resolveTeam(client, id);
        const members = await t.members();
        const estimateConfig = {
          type: t.issueEstimationType,
          allowZero: t.issueEstimationAllowZero,
          extended: t.issueEstimationExtended,
        };
        const validEstimates =
          t.issueEstimationType !== "notUsed" ? getValidEstimates(estimateConfig) : null;
        printJson({
          id: t.id,
          name: t.name,
          key: t.key,
          description: t.description,
          estimates: {
            type: t.issueEstimationType,
            allowZero: t.issueEstimationAllowZero,
            extended: t.issueEstimationExtended,
            default: t.defaultIssueEstimate,
            validValues: validEstimates,
            display: validEstimates
              ? formatEstimateScale(t.issueEstimationType, validEstimates)
              : null,
          },
          members: members.nodes.map((m) => ({
            id: m.id,
            name: m.name,
            email: m.email,
          })),
        });
      } catch (err) {
        printError(err instanceof Error ? err.message : "Get failed");
      }
    });
}

File: internal/cli/team/get.go
package team

import (
	"context"

	"github.com/Khan/genqlient/graphql"
	"github.com/spf13/cobra"

	"github.com/shhac/lin/internal/cli/shared"
	apierrors "github.com/shhac/lin/internal/errors"
	"github.com/shhac/lin/internal/estimates"
	"github.com/shhac/lin/internal/linear"
	"github.com/shhac/lin/internal/resolvers"
)

func registerGet(team *cobra.Command) {
	cmd := &cobra.Command{
		Use:   "get <id>...",
		Short: "Team details and members",
		Args:  cobra.MinimumNArgs(1),
		RunE: func(_ *cobra.Command, args []string) error {
			return shared.GetEntities(args, func(client graphql.Client, id string) (any, error) {
				resolved, err := resolvers.ResolveTeam(client, id)
				if err != nil {
					return nil, err
				}

				resp, err := linear.TeamGet(context.Background(), client, resolved.ID)
				if err != nil {
					return nil, apierrors.ClassifyGraphQLError(err)
				}
				t := resp.Team

				var validValues []int
				var display *string
				if t.IssueEstimationType != "notUsed" {
					cfg := estimates.BuildConfig(t.IssueEstimationType, t.IssueEstimationAllowZero, t.IssueEstimationExtended)
					validValues = estimates.ValidEstimates(cfg)
					d := estimates.FormatScale(t.IssueEstimationType, validValues)
					display = &d
				}

				members := make([]map[string]any, len(t.Members.Nodes))
				for i, m := range t.Members.Nodes {
					members[i] = map[string]any{
						"id":    m.Id,
						"name":  m.Name,
						"email": m.Email,
					}
				}

				return map[string]any{
					"id":          t.Id,
					"name":        t.Name,
					"key":         t.Key,
					"description": t.Description,
					"estimates": map[string]any{
						"type":        t.IssueEstimationType,
						"allowZero":   t.IssueEstimationAllowZero,
						"extended":    t.IssueEstimationExtended,
						"default":     t.DefaultIssueEstimate,
						"validValues": validValues,
						"display":     display,
					},
					"members": members,
				}, nil
			})
		},
	}
	team.AddCommand(cmd)
}
